benchmarks/README.md

```diff
@@ -0,0 +1,64 @@
+# sgkit Benchmarks
+
+Benchmarking sgkit with Airspeed Velocity.
+
+## Usage
+
+Airspeed Velocity manages building the environment via conda itself. The recipe for the same in
+defined in the `asv.conf.json` configuration file.
+
+* To run the benchmark suite:
+
+```bash
+asv run --config benchmarks/asv.conf.json
+```
+
+## Benchmark Results
+
+The benchmark results are uploaded to the [benchmarks repository][https://github.com/pystatgen/sgkit-benchmarks] via
+`benchmark.yml` Github Actions.
+
+They can be viewed locally by running the following commands:
+
+
+* Generate html
+
+```bash
+asv publish --config benchmarks/asv.conf.json -v
+```
+
+* Run local server
+
+```bash
+asv preview --config benchmarks/asv.conf.json -v
+```
+
+The benchmarks over time can be seen here: https://pystatgen.github.io/sgkit-benchmarks
+
+
+## Writing benchmarks
+
+Benchmarks should be written in the `benchmarks/` directory as `.py` files. For more information on different
+types of benchmarks see the documentation here: https://asv.readthedocs.io/en/stable/writing_benchmarks.html#writing-benchmarks
+
+
+## Benchmark Machine
+
+The benchmark machine is the Github Actions machine, which has roughly the following configurations:
+
+```json
+{
+    "arch": "x86_64",
+    "cpu": "Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz",
+    "machine": "fv-az183-669",
+    "num_cpu": "2",
+    "os": "Linux 5.4.0-1039-azure",
+    "ram": "7121276",
+    "version": 1
+}
```

Comment:
How did you discover this configuration? It might be useful to add a comment here explaining that.

It is possible to log the machine spec that each run took place on?

Reply:
This is basically spitted out while running `asv machine` command. The problem I had while trying to log machine spec for each run is the machine name should be same for the graphs to be meaningful, otherwise it will be points for each machine. I can explore to log it anyways though, without affecting the graphs.

Reply:
I have added the logging of machine configuration of the run in the CI, did you meant saving that in the results repository?

Reply:
Ideally yes, but this is a good start if that's not possible or easy to do.

Reply:
I think it is possible, as in we can somehow add a key in the results json here (assuming `asv` ignores it):
https://github.com/aktech/sgkit-benchmarks/blob/655317b0d50a52f85b4c43e81b00a4a72d76a0ed/results/github-actions/2196f390-conda-py3.8.json#L1

What do you think?